y/y.go

```diff
@@ -144,6 +143,12 @@ func SameKey(src, dst []byte) bool {
 	if len(src) != len(dst) {
 		return false
 	}
+	srcKey := ParseKey(src)
+	dstKey := ParseKey(dst)
+	lastIdx := len(srcKey) - 1
+	if srcKey[lastIdx] != dstKey[lastIdx] {
```

Comment:
Interesting trick.

Reply:
Then we have to pay the cost to check if the length is larger than 8, the benefit is (1/256) chance to avoid a bytes comparison, seems not worth it.